### PR TITLE
feature: add status colors

### LINKED ISSUE #188: Event Color Coding + Statuses on list view
Events should be color coded on the calendar, and their statuses should be shown on the list view.

### PR DESCRIPTION
**Pull request by yuw006**
closes #188

### CODE REVIEW

**Comment by yuw006** [comment]
I used [Resources](https://devexpress.github.io/devextreme-reactive/react/scheduler/docs/reference/resources/)

File: frontend/src/pages/CalendarPage/index.js
Comment (thaigillespie) [inline frontend/src/pages/CalendarPage/index.js:134]: Since text is an optional field (https://devexpress.github.io/devextreme-reactive/react/scheduler/docs/reference/resources/#resourceinstance) you don't have to include it here in these JS objects. If you do want to include text, I think doing something like `text: 'Readied Event'` and `'text: Pending Event'` would be a lot better.

**Review by thaigillespie — CHANGES_REQUESTED** [review]
Overall lgtm. I haven't used Resources so I'm not the best judge at whether or not you're using it well here (but the code looks right and your screenshots indicate that it works so it's good to me). Just had one small comment and I'll approve when you're done resolving it. 

Also, if possible, can you upload some photos of how the status colors of the events look with the whole calendar page in the description of your PR?

File: frontend/src/pages/CalendarPage/index.js
Comment (godwinpang) [inline frontend/src/pages/CalendarPage/index.js:153]: For this block could just have 1 ternary statement? view==='calendar' can just control all of the 3 FormControlLabels.

File: frontend/src/pages/CalendarPage/index.js
Comment (godwinpang) [inline frontend/src/pages/CalendarPage/index.js:126]: What do you think about having this within the Calendar component instead of being in the CalendarPage? I think there's merit in doing so because I don't see CalendarPage as being responsible for controlling *how* the Calendar component should color the events.

Specifically, resources is of an object type that's directly from the dx-scheduler component. Having to define resources outside of the Calendar component tightly couples our CalendarPage to our *implementation* of the Calendar component. For example, if we wanted to switch away from using dx-scheduler in the future, we'd have to change both the CalendarPage and the Calendar component - this means that the abstraction that the Calendar component provides is a "leaky" one.

File: frontend/src/pages/CalendarPage/index.js
Comment (godwinpang) [reply frontend/src/pages/CalendarPage/index.js:134]: +1 here - I agree with Thai; probably would be nicer to leave out text, not sure what its purpose is here as of now

**Review by godwinpang — CHANGES_REQUESTED** [review]

**Comment by yuw006** [comment]
I fixed the the resources field, and placement of the variable. Simplified the code to hide the check boxes. Here are the screenshots:
Calendar View:
image
List View:
image

File: frontend/src/pages/CalendarPage/components/Calendar/index.js
Comment (thaigillespie) [inline frontend/src/pages/CalendarPage/components/Calendar/index.js:24]: I appreciate the change to hex colors!

File: frontend/src/pages/CalendarPage/components/Calendar/index.js
Comment (thaigillespie) [inline frontend/src/pages/CalendarPage/components/Calendar/index.js:33]: Based on how you defined `instances` starting from line 21, the proptypes for it should be an array of objects, and not just an object

**Review by thaigillespie — CHANGES_REQUESTED** [review]
Overall lgtm, just one small thing that needs changing

File: frontend/src/pages/CalendarPage/components/Calendar/index.js
Comment (thaigillespie) [reply frontend/src/pages/CalendarPage/components/Calendar/index.js:33]: Well actually, since resources is not even a prop of the Calendar component, you don't even need these propTypes definitions. Can probably just remove them. One day most of our files will be changed to TypeScript and we'll have a better time with types : )

**Review by godwinpang — APPROVED** [review]
lgtm - @thaigillespie waiting on ur review then @yuw006 feel free to squash and merge :)

**Review by thaigillespie — APPROVED** [review]
Lgtm @yuw006 merge whenever you're free :)

**Pull request merged**
